packages/core/src/delegation/Delegation.ts

```diff
@@ -36,13 +37,17 @@ export default abstract class DelegationBaseNode
    *
    * @param id The unique identifier of the delegation node.
    * @param account The owner address of the delegation node.
+   * @param revoked
    */
   public constructor(
     id: IDelegationBaseNode['id'],
-    account: IDelegationBaseNode['account']
+    account: IDelegationBaseNode['account'],
+    revoked: IDelegationBaseNode['revoked']
   ) {
+    errorCheck(id, account, revoked)
     this.account = account
     this.id = id
+    this.revoked = revoked
```

Comment:
idea: let `errorCheck` accept an `IDelegationBaseNode` as its only argument, then call `errorCheck(this)` after assignment. Think that would work? Makes more sense in my head to be able to check an `IDelegationBaseNode` object.
```suggestion
    this.account = account
    this.id = id
    this.revoked = revoked
    errorCheck(this)
```

Reply:
update: if we would also be changing the base node constructor signature to work similar to the two child classes, we may just call the `errorCheck` on the input object

Reply:
Not sure I understand what you mean with the second back. I have updated to accept `this`.